tests/integration/s2n_dynamic_record_size_test.py

```diff
@@ -200,54 +211,53 @@ def test(host, port, test_ciphers, threshold):
 
     return failed
 
-def analyze_latency_dump(array):
-    failed = 0
+
+def analyze_tcp_dump(array, threshold):
+    """
+    This function iterates though each line of the TCP dump and reads the length of each segment, to check that it is
+    the correct size. It keeps account the total bytes_transferred and therefore tests that all segment sizes before
+    the threshold is met are less than the maximum segment size (mss). Once this threshold has been met, it tests
+    that the segment size has been increased by verifying that at least one segment is greater than 1500B.
```

Comment:
One last thing I didn't think about until you wrote this awesome comment: should this test actually test that **all** packets after the threshold are large? My understanding from https://github.com/awslabs/s2n/blob/main/docs/USAGE-GUIDE.md#s2n_connection_prefer_throughputstruct-s2n_connection-conn is once you're in throughput mode you stay in it unless there is no traffic and you hit the timeout. Nothing in this test introduces a delay so s2n will stay in throughput mode.

Reply:
It's a good question, I kept the functionality of the test the same as it was in when it was originally written by @fatrat1117 as I didn't want to change the scope of the test. If you look at an example TCP dump we almost always see a packet of size 400-900B a packet or two after the threshold has been met and high throughput mode is on:

![Screenshot 2020-09-17 at 11 26 10](https://user-images.githubusercontent.com/68056884/93458772-a9959b80-f8d8-11ea-84ba-3b5bf9a00d3a.png)

Depending on the cipher suite tested, there can be a few packets around this size even in high throughput mode - perhaps it's due to some messages in the handshake simply not needing to utilize the increased maximum segment size? Of course the final finished message packet will also be 0, so we wouldn't want to enforce all packets are large.

Reply:
That makes sense.